src/plrust.rs

```diff
@@ -103,11 +80,20 @@ pub(crate) fn compile_function(fn_oid: pg_sys::Oid) -> eyre::Result<(PathBuf, Ou
 
     let generated = unsafe { UserCrate::try_from_fn_oid(db_oid, fn_oid)? };
     let provisioned = generated.provision(&work_dir)?;
-    let (built, output) = provisioned.build(&work_dir, pg_config, Some(target_dir.as_path()))?;
-
+    let crate_dir = provisioned.crate_dir().to_path_buf();
+    let (built, output) = provisioned.build(pg_config, target_dir.as_path())?;
     let shared_object = built.shared_object();
 
-    Ok((shared_object.into(), output))
+    // store the shared object in our table
+    plrust_proc::create_or_replace_function(fn_oid, shared_object)?;
+
+    // cleanup after ourselves
+    tracing::trace!("removing {}", shared_object.display());
+    std::fs::remove_file(shared_object)?;
+    tracing::trace!("removing {}", crate_dir.display());
+    std::fs::remove_dir_all(crate_dir)?;
```

Comment:
I think we shouldn't simply `?` out here but rather we should prefer to proceed (thus ignoring the errors) if we can't clean up, but still report the error occurred in tracing. We "have what we want".

Reply:
I'm not convinced that's a good idea.  We're running inside a database that likes to fail fast when something isn't as expected.  Not being able to delete a file we just created would most surely indicate some kind of terrible problem.  Waving that away doesn't seem like a nice thing to do for the user or for Postgres.

I get that it's "okay" as far as we're concerned, but I don't feel like plrust is a good arbiter of making these kinds of decisions for postgres or users.

Reply:
Hmm, okay. In that case, maybe we should annotate these with `.wrap_err("failed during function cleanup")?` to provide context on the error?